src/common/operator/cast_operators.cpp

```diff
@@ -1601,8 +1602,14 @@ struct DecimalCastOperation {
 		}
 		state.digit_count++;
 		if (NEGATIVE) {
+			if (state.result < (NumericLimits<typename T::type_t>::Minimum() / 10)) {
+				return false;
+			}
 			state.result = state.result * 10 - digit;
 		} else {
+			if (state.result > (NumericLimits<typename T::type_t>::Maximum() / 10)) {
```

Comment:
Now that the template parameter is renamed we should be able to get rid of the `type_t`, right? :)

Reply:
T is here the state, which is a DecimalCastData struct, not a numeric type. And i want the limit of the type with which this struct has been initially templated, so putting T here wouldn't work. I could perhaps add a type template to DecimalCastOperation, or maybe you have some ideas? I saw elsewhere in the code that 'using' was applied for this situation, but I figured typedef is just a simpler alternative for that.

Reply:
e.g I could emulate this
// Cast String -> Numeric

```
template <typename T>
struct IntegerCastData {
	using Result = T;
	Result result;
	bool seen_decimal;
};
struct IntegerCastOperation {
	template <class T, bool NEGATIVE>
	static bool HandleDigit(T &state, uint8_t digit) {
		using result_t = typename T::Result;
		if (NEGATIVE) {
			if (state.result < (NumericLimits<result_t>::Minimum() + digit) / 10) {
				return false;
			}

Reply:
You should be able to put `TYPE` here directly now, no?

Reply:
Thijs helped me to take a look at this and suggested that I use decltype(state.result) instead, would that be ok?

Reply:
TYPE is set in a different scope, so I can't use it directly here

Reply:
Ah apologies, yes you are correct. The github view hiding the code in the middle hid the scoping from me. The way it is set up now is fine.